src/twisted/conch/test/test_keys.py

```diff
@@ -772,10 +815,14 @@ def test_fromPrivateBlobECDSA(self):
         """
         A private EC key is correctly generated from a private key blob.
         """
+        from cryptography.hazmat.primitives.asymmetric import ec
+        publicNumbers = ec.EllipticCurvePublicNumbers(
+            x=keydata.ECDatanistp256['x'], y=keydata.ECDatanistp256['y'],
+            curve=ec.SECP256R1())
         ecblob = (
             common.NS(keydata.ECDatanistp256['curve']) +
-            common.MP(keydata.ECDatanistp256['x']) +
-            common.MP(keydata.ECDatanistp256['y']) +
+            common.NS(keydata.ECDatanistp256['curve'][-8:]) +
+            common.NS(publicNumbers.encode_point()) +
```

Comment:
I don't understand why these two are equivalent. Can you elaborate what's going on here?

Reply:
To elaborate slightly: as far as I can tell, the previous ECDSA private blob decoder in `_fromString_PRIVATE_BLOB` can never have been used successfully, since it essentially existed only to decode `SSH_AGENTC_ADD_IDENTITY` messages and it used an incorrect format for those.  This was masked by correspondingly incorrect tests, as shown here.

When I expanded the scope of `_fromString_PRIVATE_BLOB` so that it's now also used to decode new-format private keys and tried to test it against real keys generated by OpenSSH, I discovered this discrepancy.  I considered whether Twisted's compatibility guidelines obliged me to keep the old implementation around somehow.  However, I really think this is just a straight bug-fix for code that wasn't properly interop-tested when it was first introduced, probably because it came along with various other changes for ECDSA support and this particular bit of it was overlooked.

It's somewhat unfortunate that these tests are written in such a way that it's easy for the code-under-test and the tests to be wrong in the same way; but the test changes in 4b6a2a2d85bc1493d8d0d1eb90a3896ac220da5c make it harder to get this wrong in future, because we're now testing Twisted's ability to decode new-format OpenSSH private keys generated by `ssh-keygen`, and that indirectly provides more realistic test data for `_fromString_PRIVATE_BLOB`, including its ECDSA case.